In [1]:
df = pd.read_csv("DS1_C8_Computers_Data_Project.csv")
df.head()

NameError: name 'pd' is not defined

In [ ]:
# TASK - 1 Advertising Budget for Premium Computers

# Given
pop_mean = 221.3   
sam_mean = 222.2     
std = 74.83     
n = 40
alpha = 0.05

z_test = (sam_mean - pop_mean) / (std / np.sqrt(n))

z_crit = norm.ppf(1 - alpha)

p_value = 1 - norm.cdf(z_test)

print(f"Sample mean (2023): {sam_mean}")
print(f"Population mean (2022): {pop_mean}")
print(f"Z statistic: {z_test:.4f}")
print(f"Z critical (α={alpha}): {z_crit:.4f}")
print(f"p-value (right-tailed): {p_value:.4f}")

if z_test > z_crit:
    print("Reject H0: evidence that 2023 mean is higher than 2022.")
else:
    print("Fail to reject H0: not enough evidence that 2023 mean is higher.")

Sample mean (2023): 222.2
Population mean (2022): 221.3
Z statistic: 0.0761
Z critical (α=0.05): 1.6449
p-value (right-tailed): 0.4697
Fail to reject H0: not enough evidence that 2023 mean is higher.


## **TASK 1 – Advertising Budget for Premium Computers**

**Test Type:** One-sample **Z-test**

**Hypotheses:**
- H₀: μ₍₂₀₂₃₎ ≤ 221.3 (No increase in 2023 advertising budget)
- H₁: μ₍₂₀₂₃₎ > 221.3 (2023 advertising budget is higher)

**Result Interpretation:**
- The test compares the **mean advertising budget of 2023 (222.2)** with the **population mean of 2022 (221.3)**.
- If the computed **Z-statistic > Z-critical (1.645)** and **p-value < 0.05**, we **reject H₀**.
- This means there is **statistical evidence** that the **2023 advertising budget increased**.
- Otherwise, we **fail to reject H₀**, implying no significant rise.

**Business Insight:**
The marketing spend for premium computers did **not significantly increase** from 2022 to 2023 unless the Z-test showed strong enough evidence.  
This suggests budget stability or similar ad investment levels.


In [ ]:
# TASK - 2 Price Between With & Without CD Player

with_cd = df[df['cd'] == 'yes']['price']
without_cd = df[df['cd'] == 'no']['price']

t_stat, p_value = ttest_ind(with_cd, without_cd, equal_var=False)

alpha = 0.05

print(f"T-statistic: {t_stat:.3f}")
print(f"P-value: {p_value:.3f}")

if p_value < alpha:
    print("✅ Reject H₀: There is a significant difference in average price.")
else:
    print("❌ Fail to Reject H₀: No significant difference in average price.")

T-statistic: 16.075
P-value: 0.000
✅ Reject H₀: There is a significant difference in average price.


## **TASK 2 – Price Difference Between Computers With & Without CD Player**

**Test Type:** **Independent two-sample t-test** (unequal variances)

**Hypotheses:**
- H₀: μ₁ = μ₂ (Average prices are equal)
- H₁: μ₁ ≠ μ₂ (Average prices differ)

**Result Interpretation:**
- Two groups: computers **with CD** vs **without CD**.
- If **p-value < 0.05**, we **reject H₀**, concluding that **having a CD drive affects price**.
- Otherwise, we fail to reject H₀, meaning **CD presence doesn’t significantly change price**.

**Business Insight:**
If a difference exists, CD drives might be considered a **price-influencing feature**, potentially reflecting consumer or market valuation.  
If not, it indicates **CD drives are no longer a premium feature** in modern PCs.


In [ ]:
# TASK - 3.1

premium_df = df[df['premium'] == 'yes']

sample_premium = premium_df.sample(n=25, random_state=1)

t_stat, p_value = ttest_1samp(sample_premium['price'], 2200)

alpha = 0.05

print(f"Sample Mean: {sample_premium['price'].mean():.2f}")
print(f"T-statistic: {t_stat:.3f}")
print(f"P-value: {p_value:.3f}")

if p_value < alpha:
    print("✅ Reject H₀: The mean price of premium computers differs significantly from $2200.")
else:
    print("❌ Fail to Reject H₀: No significant difference from $2200.")

Sample Mean: 2214.28
T-statistic: 0.098
P-value: 0.923
❌ Fail to Reject H₀: No significant difference from $2200.


## **TASK 3.1 – Premium Computer Prices vs $2200 Benchmark**

**Test Type:** **One-sample t-test**

**Hypotheses:**
- H₀: μ = 2200 (Mean price equals $2200)
- H₁: μ ≠ 2200 (Mean price differs from $2200)

**Result Interpretation:**
- A **sample of 25 premium computers** is compared to a benchmark price of **$2200**.
- If **p-value < 0.05**, reject H₀ → premium computers are **priced significantly different** from $2200.
- Otherwise, fail to reject H₀ → no significant difference from $2200.

**Business Insight:**
If prices differ, the company should **reconsider its pricing strategy** for premium products—either they are **undervalued or overpriced** compared to market expectations.


In [ ]:
# TASK - 3.2

premium = df[df['premium'] == 'yes']['price']
non_premium = df[df['premium'] == 'no']['price']

t_stat2, p_value2 = ttest_ind(premium, non_premium, equal_var=True)

print(f"T-statistic: {t_stat2:.3f}")
print(f"P-value: {p_value2:.3f}")

if p_value2 < alpha:
    print("✅ Reject H₀: There is a significant difference in mean prices between premium and non-premium computers.")
else:
    print("❌ Fail to Reject H₀: No significant difference in mean prices between premium and non-premium computers.")


T-statistic: -6.404
P-value: 0.000
✅ Reject H₀: There is a significant difference in mean prices between premium and non-premium computers.


## **TASK 3.2 – Premium vs Non-Premium Computer Prices**

**Test Type:** **Independent two-sample t-test** (equal variances assumed)

**Hypotheses:**
- H₀: μ₁ = μ₂ (Mean prices are the same)
- H₁: μ₁ ≠ μ₂ (Mean prices differ)

**Result Interpretation:**
- Compares **premium** vs **non-premium** computer prices.
- If **p-value < 0.05**, reject H₀ → there’s a **significant difference** in mean prices.
- Otherwise, fail to reject H₀ → both categories are **priced similarly**.

**Business Insight:**
A significant difference validates the **premium brand positioning**, confirming that premium models command a **price premium**.  
If not significant, it indicates **pricing overlap**, which could confuse market perception.
